presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -211,6 +199,31 @@ else if (state.isDone()) {
         }
     }
 
+    private synchronized void registerDynamicFilteringQuery()
+    {
+        if (!isEnableDynamicFiltering(stateMachine.getSession())) {
+            return;
+        }
+
+        if (isDone()) {
```

Comment:
Maybe merge these conditions

Reply:
I prefer to leave comment here as it's not obvious why we check this condition here. I've improved comment.